### PR TITLE
docs: Add issue template

### PR DESCRIPTION
**Pull request by tanmay-pnaik**
## Overview

Add issue template

## Notes for reviewer

Adapted from https://github.com/bank-vaults/vault-operator/tree/main/.github/ISSUE_TEMPLATE

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      placeholder: 1.27.0
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Kubernetes Distribution/Provisioner
+      description: Which Kubernetes distribution/privisioner are you using?
+      placeholder: e.g. GKE, EKS, AKS etc
+    validations:
+      required: true
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:62]:
```suggestion
```

This information does not seem relevant for the SDK.

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        apiVersion: "vault.banzaicloud.com/v1alpha1"
+        kind: "Vault"
+        metadata:
+          name: "vault"
+        # ...
+  - type: textarea
+    attributes:
+      label: Logs
+      description: Operator application logs (if relevant).
+      render: shell
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:99]:
```suggestion
```

Same as above

**Review by sagikazarmark — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,103 @@
+name: 🐛 Bug report
+description: Report a bug to help us improve the Vault SDK
+labels: [bug]
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:3]:
```suggestion
labels: [kind/bug]
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        If you are not sure if it is a bug or not, you can contact us via the available [support channels](https://github.com/bank-vaults/vault-sdk/issues/new/choose).
+  - type: checkboxes
+    attributes:
+      label: Preflight Checklist
+      description: Please ensure you've completed all of the following.
+      options:
+        - label: I have searched the [issue tracker](https://www.github.com/bank-vaults/vault-sdk/issues) for an issue that matches the one I want to file, without success.
+          required: true
+        - label: I am not looking for support or already pursued the available [support channels](https://github.com/bank-vaults/vault-sdk/issues/new/choose) without success.
+          required: true
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:21]:
```suggestion
          required: true
        - label: I agree to follow the [Code of Conduct](https://bank-vaults.dev/docs/code-of-conduct/).
          required: true
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        Please describe what you would like to change/add and why in detail by filling out the template below.
+
+        If you are not sure if your request fits into Dex, you can contact us via the available [support channels](https://github.com/bank-vaults/vault-sdk/issues/new/choose).
+  - type: checkboxes
+    attributes:
+      label: Preflight Checklist
+      description: Please ensure you've completed all of the following.
+      options:
+        - label: I have searched the [issue tracker](https://www.github.com/bank-vaults/vault-sdk/issues) for an issue that matches the one I want to file, without success.
+          required: true
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:18]:
```suggestion
```suggestion
          required: true
        - label: I agree to follow the [Code of Conduct](https://bank-vaults.dev/docs/code-of-conduct/).
          required: true
```
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+name: 🎉 Feature request
+description: Suggest an idea for the Vault SDK
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:2]:
```suggestion
description: Suggest an idea for the Vault SDK
labels: [kind/enhancement]
```

**Review by sagikazarmark — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/config.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+blank_issues_enabled: true
+contact_links:
+  - name: 📖 Documentation enhancement
+    url: https://github.com/bank-vaults/bank-vaults.dev/issues
+    about: Suggest an improvement to the documentation
+
+  - name: 📚 Documentation
+    url: https://bank-vaults.dev/docs/go-library/
+    about: Check the documentation for help
```
Comment (ramizpolic) [inline .github/ISSUE_TEMPLATE/config.yml:9]:
```suggestion
  - name: 📚 Documentation
    url: https://bank-vaults.dev/docs/go-library/
    about: Check the documentation for help
    
  - name: 📖 Documentation enhancement
    url: https://github.com/bank-vaults/bank-vaults.dev/issues
    about: Suggest an improvement to the documentation
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+        Please describe what you would like to change/add and why in detail by filling out the template below.
+
+        If you are not sure if your request fits into Dex, you can contact us via the available [support channels](https://github.com/bank-vaults/vault-sdk/issues/new/choose).
+  - type: checkboxes
+    attributes:
+      label: Preflight Checklist
+      description: Please ensure you've completed all of the following.
+      options:
+        - label: I have searched the [issue tracker](https://www.github.com/bank-vaults/vault-sdk/issues) for an issue that matches the one I want to file, without success.
```
Comment (ramizpolic) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:18]:
```suggestion
        - label: I have checked the [issue tracker](https://www.github.com/bank-vaults/vault-sdk/issues) and found no issue that matches the one I want to submit.
```

**Review by ramizpolic — CHANGES_REQUESTED** [review]

**Comment by ramizpolic** [comment]
Much appreciated @tanmay-pnaik, this is awesome! Would you mind fixing the pipeline issues? I can merge it afterward.

**Comment by tanmay-pnaik** [comment]
> Much appreciated @tanmay-pnaik, this is awesome! Would you mind fixing the pipeline issues? I can merge it afterwards.

Done!

**Review by ramizpolic — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/config.yml [hunk 9]
```diff
@@ -3,7 +3,6 @@ contact_links:
   - name: 📚 Documentation
     url: https://bank-vaults.dev/docs/go-library/
     about: Check the documentation for help
-
```
[the remark this answers is not in the document]
Comment (ramizpolic) [reply .github/ISSUE_TEMPLATE/config.yml]: The lint check was failing due to trailing spaces. I removed it via GitHub UI. AFAIK There is no need for more than one person to sign the commit. 

Also, similar changes should be added to other PRs. This is very useful for the project as a whole, so thanks again @tanmay-pnaik 🚀

[the remark this answers is not in the document]
Comment (tanmay-pnaik) [reply .github/ISSUE_TEMPLATE/config.yml]: @ramizpolic Have already created similar PRs across the project! :grin:

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**